### PR TITLE
Add v1alpha1 API package

### PR DESCRIPTION
**Pull request by pmorie**
Just adding a new types.go file for the `v1alpha1` package so we can begin working on getting code generation going.

### CODE REVIEW

**Comment by arschles** [comment]
LGTM

**Comment by MHBauer** [comment]
lgtm

**Comment by duglin** [comment]
LGTM

File: pkg/apis/servicecatalog/v1alpha1/types.go [hunk 1]
```diff
@@ -0,0 +1,264 @@
...
+)
+
+// BrokerSpec represents a description of a Broker.
+type BrokerSpec struct {
+	// The URL to communicate with the Broker via..
+	URL string
+
+	// Auth credentials should live in an api.Secret that
+	// is documented to have "username" and "password" keys
+	AuthUsername string
```
Comment (deads2k) [inline pkg/apis/servicecatalog/v1alpha1/types.go]: @pmorie is there an issue for this?  Having it embedded in an API object here where it's always available and never out of sync will significantly (possibly vastly) simplify controller logic that has to compensate for it and hides API access problems (who has access across all secrets?).  This is a very dangerous thing to have.

Comment (deads2k) [inline pkg/apis/servicecatalog/v1alpha1/types.go]: @smarterclayton @liggitt who have seen APIs like this go wrong too and probably have a vague interest.

**Pull request merged**